File: app/src/main/java/com/gmail/fattazzo/formula1world/ergast/imagedb/objects/Season.java
package com.gmail.fattazzo.formula1world.ergast.imagedb.objects;

import com.activeandroid.Model;
import com.activeandroid.annotation.Column;
import com.activeandroid.annotation.Table;
import com.gmail.fattazzo.formula1world.domain.F1Season;

/**
 * @author fattazzo
 *         <p/>
 *         date: 05/06/17
 */
@Table(name = "seasons")
public class Season extends Model {

    @Column
    public String url;

    @Column
    public String description;

    public F1Season toF1Season() {
        F1Season f1Season = new F1Season();
        f1Season.year = getId().intValue();
        f1Season.url = url;
        f1Season.description = description;
        f1Season.current = false;
        return f1Season;
    }

    @Override
    public String toString() {
        return "Season{" +
                "url='" + url + '\'' +
                ", description='" + description + '\'' +
                '}';
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/ergast/imagedb/objects/Season.kt
/*
 * Project: total-gp-world
 * File: Season.kt
 *
 * Created by fattazzo
 * Copyright © 2018 Gianluca Fattarsi. All rights reserved.
 *
 * MIT License
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.gmail.fattazzo.formula1world.ergast.imagedb.objects

import com.activeandroid.Model
import com.activeandroid.annotation.Column
import com.activeandroid.annotation.Table
import com.gmail.fattazzo.formula1world.domain.F1Season

/**
 * @author fattazzo
 *
 *
 * date: 05/06/17
 */
@Table(name = "seasons")
class Season : Model() {

    @Column
    var url: String? = null

    @Column
    var description: String? = null

    fun toF1Season(): F1Season {
        val f1Season = F1Season()
        f1Season.year = id!!.toInt()
        f1Season.url = url
        f1Season.description = description
        f1Season.current = false
        return f1Season
    }

    override fun toString(): String {
        return "Season{" +
                "url='" + url + '\'' +
                ", description='" + description + '\'' +
                '}'
    }
}
